lib/srv/alpnproxy/local_proxy.go

```diff
@@ -205,14 +230,27 @@ func (l *LocalProxy) GetAddr() string {
 func (l *LocalProxy) handleDownstreamConnection(ctx context.Context, downstreamConn net.Conn) error {
 	defer downstreamConn.Close()
 
+	var certs []tls.Certificate
+	shouldTunnel, downstreamConn, err := l.shouldTunnelConn(ctx, downstreamConn)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	if shouldTunnel {
+		l.cfg.Log.Debug("Loading client certs to dial remote proxy.")
+		certs, err = l.getCerts()
+		if err != nil {
+			return trace.Wrap(err)
+		}
+	}
```

Comment:
prior to this change, the only thing that determined if certs were used to dial upstream (i.e. like `--tunnel`) was whether certs were set in the local proxy. Since we wanted to support non-tunnel postgres local proxy `tsh proxy db postgres`, I had to load certs from the profile in case tunnel was required for a cancel request. But then certs were loaded into local proxy but should only be used for cancel requests. So yes it's needed due to that requirement